### DISCUSSION TITLE
Is it possible to convert an "Apple Virtualisation" image to "QEMU"

### DISCUSSION THREAD

**Discussion by PicoCreator** [utmapp/UTM#3784] [UTM Help (macOS)]
1) Is the "Apple Virtualisation" mode in UTM using QEMU underneath the hood - or simply bypassing it and integrating straight to apple code.

2) Is it possible to convert an existing "Apple Virtualisation" image to a QEMU image - if so what are the process to do so? 

Side note: Damn the "Apple Virtualisation" performance was surprisingly good, and I would like to do some apple to apple benchmarks against QEMU. Especially on GPU related workloads

**Comment by iustin** [comment]
I would be interested as well on this, also on Qemu→Apple conversion. The disk contents shouldn't be impacted, but maybe needs conversion from qcow2 to something else.

**Reply by crasm** [reply]
I was able to convert to raw format successfully.
```
$ cd Library/Containers/com.utmapp.UTM/Data/Documents
$ qemu-img convert -f qcow2 -O raw "Linux.utm/Data/<qemu-disk-UUID>.qcow2" "Linux 3.utm/Data/<hvf-disk-UUID>.raw"
```

My qcow2 image was sized for up to 64GB but the converted raw image was only the 12GB or so I already had. Latency, performance, and graphics are much better, but it ended up for naught in my case because USB passthrough isn't supported.

**Comment by mikemallin** [comment]
I can't answer #1, but it is possible to convert a HVF raw image using qemu-img:

`qemu-img convert -f raw -O qcow2 <Path/To/HVF/Disk.img> ~/output-img.qcow2`

I'd suspect the reverse should also hold to go from Apple to QEMU.

**Comment by RonGros** [comment]
Did you ever test that? does it work?

**Reply by mikemallin** [reply]
I've only tested from img to qcow2 using a dev vm on my laptop. I haven't tried the reverse.

**Comment by proski** [comment]
Conversion from RAW only works is "Use Apple Sparse Format" was unchecked when the image was created. If the image is in sparse format, it cannot be converted by qemu-img (the conversion would produce an invalid image). One way to deal with it is to add an uncompressed image of the same size and a bootable ISO to the VM, boot from the ISO and copy the disk. Then make qcow2 from the uncompressed image. I tested it for a simple Alpine install.

There is also Apple's hdiutil, but it also supports uncompressed images only.
